functions/Start-DbaAgentJob.ps1

```diff
@@ -120,69 +164,72 @@ function Start-DbaAgentJob {
-                        Start-Sleep -Milliseconds $SleepPeriod
-                        $currentjob.Refresh()
+                    Write-Message -Level Verbose -Message "Sleeping for $SleepPeriod ms and refreshing"
+                    Start-Sleep -Milliseconds $SleepPeriod
+                    $currentjob.Refresh()
 
-                        # If it failed fast, speed up output
-                        if ($lastrun -ne $currentjob.LastRunDate) {
-                            $i = 600
-                        }
+                    # If it failed fast, speed up output
+                    if ($lastrun -ne $currentjob.LastRunDate) {
+                        $i = 600
                     }
+                }
 
-                    # Wait for the job
-                    if (Test-Bound -ParameterName Wait) {
-                        while ($currentjob.CurrentRunStatus -ne 'Idle') {
-                            $currentRunStatus = $currentjob.CurrentRunStatus
-                            $currentStep = $currentjob.CurrentRunStep
-                            $jobStepsCount = $currentjob.JobSteps.Count
-                            $currentStepRetryAttempts = $currentjob.CurrentRunRetryAttempt
-                            if (-not $currentStepRetryAttempts) { $currentStepRetryAttempts = "0" }
-                            $currentStepRetries = $currentjob.RetryAttempts
-                            if (-not $currentStepRetries) { $currentStepRetries = "Unknown" }
-                            Write-Message -Level Verbose -Message "$currentjob is $currentRunStatus, currently on Job Step $currentStep / $jobStepsCount, and has tried $currentStepRetryAttempts / $currentStepRetries retry attempts"
-                            Start-Sleep -Seconds $WaitPeriod
-                            $currentjob.Refresh()
-                        }
-                        Get-DbaAgentJob -SqlInstance $server -Job $currentjob.Name
-                    } else {
-                        Get-DbaAgentJob -SqlInstance $server -Job $currentjob.Name
+                if ((Test-Bound -ParameterName Wait) -and (Test-Bound -not -ParameterName Parallel)) {
+                    # Wait for the job each job in a serialized fashion.
+                    while ($currentjob.CurrentRunStatus -ne 'Idle') {
+                        Invoke-Command -ScriptBlock $waitBlock -ArgumentList @($currentjob, $true, $WaitPeriod)
                     }
+                    Get-DbaAgentJob -SqlInstance $server -Job $($currentjob.Name)
+                } elseif (Test-Bound -not -ParameterName Parallel) {
+                    Get-DbaAgentJob -SqlInstance $server -Job $($currentjob.Name)
+                }
+            }
+        }
+
+        # Wait for each job to be done in parallel
+        if (Test-Bound -ParameterName Parallel) {
+            while ($InputObject.Where( { $PSItem.CurrentRunStatus -ne 'Idle' } ).Count -gt 0) {
```

Comment:
This is PS4 syntax iirc, while we're trying to maintain the backwards compatibility with at least PS3. Can this be changed to a piped `where`?

Reply:
Yes, use of `.Where()` needs to be reverted to pipe to `Where-Object`